### PR TITLE
chore: linting, formatting and their configuration

### PR DESCRIPTION
**Pull request by CondensedMilk7**
We should probably get this out of the way before working on other issues. This is the initial setup of prettier & eslint. We could use the "airbnb" plugin or whichever you think will suit the project.

Let me know what to add/change in the configuration. I haven't set up linter and formatter together on my own before so I bet there are changes pending.

### CODE REVIEW

**Comment by arothuis** [comment]
Nice! Code style is always very personal, so it's good to pick a standard and stick with it.

My personal preference is (as you might suspect) double quotes and semicolons, but it is probably more common to just follow the airbnb style guide. This makes it easier for other potential contributors. Can you configure eslint thusly and let prettier follow suit? Plugins allowed.

**Comment by arothuis** [comment]
We may also want to add linting to our `build.yml` via `npm run lint`, i.e. before the following rules
https://github.com/arothuis/markdown-it-biblatex/blob/edc43bfe644d68c9ee318f382ba4cd9280960870/.github/workflows/build.yml#L24-L26

File: package.json [hunk 1]
```diff
@@ -18,7 +18,8 @@
   },
   "scripts": {
     "test": "mocha --reporter-option maxDiffSize=0",
-    "test:cov": "nyc npm run test"
+    "test:cov": "nyc npm run test",
+    "lint": "eslint ./src"
```
Comment (arothuis) [inline package.json:22]: We may want our tests to be linted too. Not sure if there's something stopping this?

Comment (CondensedMilk7) [reply package.json:22]: Just the "no-udef" rule, but I guess we can disable it for this file.

**Comment by CondensedMilk7** [comment]
Added airbnb configuration, which prefers single quotes so I left that in `.prettierrc`.

I also have overwritten some configurations regarding "param reassign" and "use before defined", because it seems like you are deliberately going for it at several places and I don't have a problem with that. I disabled "no-udef" rule for the test file, because the mocha cli recognizes those functions that eslint thinks are undefined.

The lint command has also been added to `build.yml`.

File: package.json (hunk 1 above)
Comment (arothuis) [reply package.json]: It might work if we add mocha and chai to the eslint env?

**Comment by arothuis** [comment]
Thanks for the update! I might use these settings for my other markdown-it plugins as well. 

Just one remark. Perhaps we can add our global dependencies (node, mocha and chai) to our env settings in eslint, so we don't need to allow undefined identifiers.

**Comment by CondensedMilk7** [comment]
Added node and mocha to the env, no more eslint errors for the test file. The chai is not recognized by eslint, but it causes no problems without it either. Let me know if I should manage that or anything else.

There is `eslint-plugin-chai-friendly` we could use.

**Comment by arothuis** [comment]
You're a legend! Thanks for your changes and patience. Will merge.

**Pull request merged**